### PR TITLE
14 added and removed

### PR DESCRIPTION
**Pull request by Sinjhin**
Solves the post part of issue #14 
Pre-processor still needs to be completed.

### CODE REVIEW

**Comment by Sinjhin** [comment]
Closes #14

File: can-define.js [hunk 1]
```diff
@@ -477,38 +477,39 @@ make = {
...
-// gets a behavior for a definition or from the defaultDefinition
-getPropDefineBehavior = function(behaviorName, prop, def, defaultDefinition) {
-	if(behaviorName in def) {
-		return def[behaviorName];
+var addDefinition = function(definition, behavior, value){
+	if(behavior === "type") {
+		var behaviorDef = value;
+		if(typeof behaviorDef === "string") {
+			behaviorDef = define.types[behaviorDef];
+			if(typeof behaviorDef === "object") {
```
Comment (justinbmeyer) [inline can-define.js]: what if behaviorDef is `null`? `prop: 2`

File: can-define.js [hunk 2]
```diff
@@ -477,38 +477,39 @@ make = {
...
-			if(behavior === "type" && typeof behaviorDef === "string") {
-				behaviorDef = define.types[behaviorDef];
-				if(typeof behaviorDef === "object") {
-					assign(definition, behaviorDef);
-					behaviorDef = behaviorDef[behavior];
-				}
-			}
-			definition[behavior] = behaviorDef;
+	
+	for(var behavior in defaultDefinition) {
```
Comment (justinbmeyer) [inline can-define.js]: Use our `each`.

Comment (justinbmeyer) [inline can-define.js]: and make `can-util/js/defaults/defaults`.

File: list/list-test.js [hunk 3]
```diff
@@ -717,3 +717,49 @@ QUnit.test("shorthand getter setter (#56)", function(){
...
+				ok(items, "items added got passed to added");
+				ok(typeof index === 'number', "index of items was passed to added and is a number");
+				ok(items[0].name === 'John', "Name was correct");
+				this.outsideProp = true;
+			},
+			removed: function(items, index){
+				removedFuncCalled = true;
+				ok(items, "items added got passed to removed");
+				ok(typeof index === 'number', "index of items was passed to removed and is a number");
+			},
```
Comment (justinbmeyer) [inline list/list-test.js]: no test for `this`

File: list/list.js [hunk 4]
```diff
@@ -100,19 +100,22 @@ var DefineList = Construct.extend("DefineList",
...
         // Make sure this is not nested and not an expando
         if (!~(""+attr).indexOf('.') && !isNaN(index)) {
-
             if (how === 'add') {
-                canEvent.dispatch.call(this, how, [newVal, index]);
+				canEvent.dispatch.call(this, how, [newVal, index]);
                 canEvent.dispatch.call(this, 'length', [this._length]);
+				if (this["*"] && typeof this["*"].added === 'function') {
+					this["*"].added.call(this, newVal, index);
+				}
```
Comment (justinbmeyer) [inline list/list.js]: check formatting

File: list/list.js [hunk 5]
```diff
@@ -100,19 +100,22 @@ var DefineList = Construct.extend("DefineList",
 
         // Make sure this is not nested and not an expando
         if (!~(""+attr).indexOf('.') && !isNaN(index)) {
-
             if (how === 'add') {
-                canEvent.dispatch.call(this, how, [newVal, index]);
+				canEvent.dispatch.call(this, how, [newVal, index]);
                 canEvent.dispatch.call(this, 'length', [this._length]);
+				if (this["*"] && typeof this["*"].added === 'function') {
+					this["*"].added.call(this, newVal, index);
```
Comment (justinbmeyer) [inline list/list.js]: `this._define.defaultProperties.added`

```
var defaultProps = this._define.defaultProperties;
var added = defaultProps.added;
if(added) {
   Observation.ignore(added).call(this, newVal, index);
}
```

Comment (justinbmeyer) [inline list/list.js]: call `added` before previous events.

**Pull request merged**